tests/e2e/creation_test.go

```diff
@@ -353,6 +360,46 @@ func (tc *testContext) validateDSC() error {
 	return nil
 }
 
+func (tc *testContext) testTrustedCABundle() error {
+	CAConfigMapName := "odh-trusted-ca-bundle"
+	CADataFieldName := "odh-ca-bundle.crt"
+
+	if tc.testDSCI.Spec.TrustedCABundle.ManagementState == operatorv1.Managed {
+		foundConfigMap := &corev1.ConfigMap{}
+		err := tc.customClient.Get(tc.ctx, client.ObjectKey{
+			Name:      CAConfigMapName,
+			Namespace: tc.testDSCI.Spec.ApplicationsNamespace,
+		}, foundConfigMap)
+
+		if err != nil {
+			return fmt.Errorf("Config map not found, %w", err)
+		}
+
+		checkNewline := strings.HasSuffix(foundConfigMap.Data[CADataFieldName], "\n")
+
+		if checkNewline == false {
+			fmt.Print("Newline not found at the end of configmap")
+		}
+
+		if strings.TrimSpace(foundConfigMap.Data[CADataFieldName]) != tc.testDSCI.Spec.TrustedCABundle.CustomCABundle {
+			return fmt.Errorf("odh-trusted-ca-bundle in config map does not match with DSCI's TrustedCABundle.CustomCABundle, needs update: %w", err)
+		}
+	} else {
+		foundConfigMap := &corev1.ConfigMap{}
+		err := tc.customClient.Get(tc.ctx, client.ObjectKey{
+			Name:      CAConfigMapName,
+			Namespace: tc.testDSCI.Spec.ApplicationsNamespace,
+		}, foundConfigMap)
+
+		if k8serr.IsNotFound(err) {
+			fmt.Printf("Config map not found in the namespace")
```

Comment:
What this whole "else" branch is supposed to test?

Reply:
so this the [check](https://github.com/opendatahub-io/opendatahub-operator/pull/1346#discussion_r1832344206), 

set managementStateChangeTrustedCA := false
check no Configmap in tc.testDSCI.Spec.ApplicationsNamespace

Reply:
Can the code be more explicit in it then?